test/exe/main_common_test.cc

```diff
@@ -304,6 +305,143 @@ TEST_P(AdminRequestTest, AdminRequestAfterRun) {
   EXPECT_EQ(1, lambda_destroy_count);
 }
 
+class MonitoredTimeSystem : public Event::SimulatedTimeSystem {
```

Comment:
nit: wouldn't these all be a bit better as mocks? gmock gives you all of this built-in...

Reply:
From my read below everything you are doing is primarily around counting calls and then doing some pass through behavior. This can be easily accomplishes with setting up an ON_CALL default, and then using EXPECT_CALL for call count verification. Ultimately it's not a big deal either way, but I would have used mocks for this.

In any case, I'm not sure it matters. If you change the constructor to require object creation do you really need these tests? Any issue would be pretty obvious via other tests.